friends/src/components/FriendList.js

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { Route } from 'react-router-dom';
+import Friend from './Friend';
+
+const FriendList = ({ friends, ...props }) => {
+  function displayForm() {
+    props.history.push('/add-friend');
+  }
+
+  return (
+    <div>
+      <ul>
+        {friends.length ? (
+          friends.map(friend => (
+            <Friend key={friend.id} friend={friend} {...props} />
+          ))
+        ) : (
+          <div>Loading...</div>
+        )}
+      </ul>
+      <Route
+        exact
+        path="/"
+        render={() => <button onClick={displayForm}>Display Form</button>}
```

Comment:
Inline function to return the new button instead of having to declare it elsewhere in the component first, nice. Alternatively, you could perform a ternary here and check to see if the `props.match.url` is strictly equal to `"/"` and return this button only if it is. The nested Route checking for an exact match within a Route with same path but _not_ exact works, but feels like an antipattern.